Refactor storage class parsing for Gateway mode

nitisht opened the pull request:
## Description
This PR updates the behaviour to print relevant error message
if storage class is set in config.json for gateway

This PR also fixes the case where storage class set via
environment variables is not parsed properly into config.json.

## Motivation and Context
This is needed to make sure the storage class field is validated only for erasure coded mode.

## How Has This Been Tested?
Manually

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

harshavardhana commented on cmd/config-current.go:
```diff
@@ -373,7 +378,8 @@ func loadConfig() error {
-	if !globalIsStorageClass {
+	// Fetch storage class from config only if it is erasure coding setup, else ignore
+	if !globalIsStorageClass && globalIsXL {
 		globalStandardStorageClass, globalRRStorageClass = globalServerConfig.GetStorageClass()
```
GetStorageClass() should return an error and must be returned during `loadConfig()`

harshavardhana commented on cmd/config-current.go:
```diff
@@ -109,8 +109,12 @@ func (s *serverConfig) SetStorageClass(standardClass, rrsClass storageClass) {
-	s.StorageClass.Standard = standardClass.Scheme + strconv.Itoa(standardClass.Parity)
-	s.StorageClass.RRS = rrsClass.Scheme + strconv.Itoa(rrsClass.Parity)
+	if standardClass.Parity != 0 {
+		s.StorageClass.Standard = standardClass.Scheme + ":" + strconv.Itoa(standardClass.Parity)
```
We should avoid this by adding a `storageClass.String()` method to get the `scheme:N`  directly this type of problem can be avoided. Then you can simply add `s.StorageClass.Standard  = standardClass.String()`

harshavardhana replied:
are you going to do these changes? @nitisht

nitisht replied:
yes @harshavardhana , @krisis had some suggestions - I am working on that, will update the PR

nitisht committed:
Refactor storage class parsing for Gateway mode

This PR updates the behaviour to print relevant error message
if storage class is set in config.json for gateway

This PR also fixes the case where storage class set via
environment variables is not parsed properly into config.json.

harshavardhana reviewed (approved):
LGTM
